src/api_manager/check_service_control.cc

```diff
@@ -76,6 +76,14 @@ void CheckServiceControl(std::shared_ptr<context::RequestContext> context,
                           << "status " << status.ToString();
         // info is valid regardless status.
         context->set_check_response_info(info);
+
+        // update consumer_project_id to service context
+        if (!info.consumer_project_id.empty()) {
+          context->set_consumer_project_id(info.consumer_project_id);
+          context->request()->AddHeaderToBackend("X-Endpoint-API-Project-ID",
```

Comment:
Which one is the right header?

X-Endpoint-API-Project-ID or X-Endpoint-API-Consumer-Project-ID

https://buganizer.corp.google.com/issues/63892801#comment5 says X-Endpoint-API-Project-ID